### PR TITLE
feat: add Dataset ACL support

### PR DESCRIPTION
**Pull request by stephaniewang526**
Fixes b/211633697

### CODE REVIEW

**Comment by stephaniewang526** [comment]
Blocked by cl/421673680. Will update the generated client version once the CL is released.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java [hunk 1]
```diff
@@ -119,6 +122,11 @@ public Type getType() {
     abstract Access toPb();
 
     static Entity fromPb(Access access) {
+      if (access.getDataset() != null) {
+        return new Dataset(
```
Comment (shollyman) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java:126]: minor constructor surprise for me, until I realized we're dealing with an ACL.Dataset and not a Dataset.  I suspect this is just local ambiguity; any other places where we need to consider the naming?

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java [hunk 2]
```diff
@@ -146,6 +154,64 @@ static Entity fromPb(Access access) {
     }
   }
 
+  /**
+   * Class for a BigQuery Dataset entity. Objects of this class represent a dataset from a different
```
Comment (shollyman) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java:158]: Maybe clarify this is a Dataset ACL entity, not a Dataset entity?

**Review by shollyman — APPROVED** [review]
Mostly my concern is naming confusion, but the precedent was set before this change made.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java (hunk 1 above)
Comment (stephaniewang526) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java]: Yeah I agree it's a little confusing so I renamed it to be more explicit: `DatasetAclEntity`.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java (hunk 2 above)
Comment (stephaniewang526) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java]: Updated. Thanks!

**Pull request merged**